Directory.Build.props

```
@@ -21,6 +21,7 @@
     <!-- <ArtifactsShippingSymbolsDir>$(ArtifactsDir)symbols\$(Configuration)\Shipping</ArtifactsShippingSymbolsDir> -->
 
     <DefineConstants Condition="'$(ContinuousIntegrationBuild)' == 'true'">$(DefineConstants);CI_BUILD</DefineConstants>
+    <DefineConstants Condition="'$(DotNetBuildFromSource)' != 'true'">$(DefineConstants);MICROSOFT_ENABLE_TELEMETRY</DefineConstants>
```

Per my understanding, with the naive build of the product, `MICROSOFT_ENABLE_TELEMETRY` will be set. Isn't that not what we want?

How about the follow?

```suggestion
    <DefineConstants Condition="'$(MicrosoftOfficialBuild)' == 'true'">$(DefineConstants);MICROSOFT_ENABLE_TELEMETRY</DefineConstants>
```

---

What is there works to opt-out of telemetry for open-source builds of .NET.
`DotNetBuildFromSource` is used throughout the different repositories to detect that.

I'm not familiar with `MicrosoftOfficialBuild`. It's not used in this repo yet.
I'll apply the suggestion.

---

`MicrosoftOfficialBuild` is something I made up. We would just need the official build to set that condition and then everything would be perfect.

Can we do that @marcpopMSFT, for the official SDK build?

---

That's more of an engineering system question as items like ContinuousIntegrationBuild are set in the scripts that live in the eng\common folder it looks like. @mmitche do you know where MicrosoftOfficialBuild would need to be set so that our official builds would include that? that probably needs to be set before this PR can go in.

---

It would need to be set in the YAML files in this repo, similar to how we enable other Microsoft specific switches in our build. e.g. https://github.com/dotnet/sdk/blob/main/.vsts-ci.yml#L34-L35

I think this will end up being a common pattern with potentially other organizations having their own logic in shared code. Perhaps we change this to something like `OfficialBuilder`, setting it to `Microsoft` for us, or perhaps `BuildOrganization`?

Thoughts @richlander?

---

Let's go for `OfficialBuilder`

---

We can always change it later if someone has a better name.

---

> It would need to be set in the YAML files in this repo, similar to how we enable other Microsoft specific switches in our build. e.g. https://github.com/dotnet/sdk/blob/main/.vsts-ci.yml#L34-L35

I've added it to this location.

> Let's go for OfficialBuilder

My personal preference was to use the existing `DotNetBuildFromSource` rather than introduce something new.